src/vault.cairo

```diff
@@ -251,5 +261,42 @@ mod Vault {
 
             self.emit(PositionProtected { token, deposit_contract, contract_owner: user, amount });
         }
+
+        /// Return liquidity from the vault by transferring tokens to the user.
+        ///
+        /// # Arguments
+        ///
+        /// * `user` - The address of the withdrawer
+        /// * `amount` - The amount of tokens to withdraw from the vault
+        ///
+        /// # Panics
+        ///
+        /// * When the caller's deposited balance is less than the withdrawal amount
+        ///
+        /// # Events
+        ///
+        /// Emits a `LiquidityReturned` event with:
+        /// * `user` - The address of the withdrawer
+        /// * `token` - The address of the withdrawn token
+        /// * `amount` - The amount of tokens withdrawn
+        fn return_liquidity(ref self: ContractState, user: ContractAddress, amount: TokenAmount){
+            let caller = get_caller_address();
+            let token = self.token.read();
+            let current_amount = self.amounts.entry(caller).read();
+            assert(current_amount >= amount, 'Not enough tokens to withdraw');
+
+            // update new amount
+            self.amounts.entry(caller).write(current_amount - amount);
```

Comment:
Why are you decreasing the balance? When tokens are returned they should be added to the user's inner balance